wrappers/llvm-mingw_toolchainfile.cmake.in

```diff
@@ -0,0 +1,83 @@
+if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.20")
+  set(_prefix "${CMAKE_CURRENT_LIST_DIR}/@RELPATH@/")
+  cmake_path(ABSOLUTE_PATH _prefix NORMALIZE)
+else()
+  get_filename_component(_prefix "${CMAKE_CURRENT_LIST_DIR}/@RELPATH@/" ABSOLUTE)
+endif()
+
+if(NOT CMAKE_SYSTEM_NAME)
+  set(CMAKE_SYSTEM_NAME Windows)
+endif()
+if(NOT CMAKE_SYSTEM_PROCESSOR)
+  set(CMAKE_SYSTEM_PROCESSOR ${CMAKE_HOST_SYSTEM_PROCESSOR})
+  if(CMAKE_HOST_WIN32)
+    if(CMAKE_SYSTEM_PROCESSOR STREQUAL "ARM64")
+      set(CMAKE_SYSTEM_PROCESSOR aarch64)
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^ARM")
+      set(CMAKE_SYSTEM_PROCESSOR armv7)
+    elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86")
+      set(CMAKE_SYSTEM_PROCESSOR i686)
+    elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
+      set(CMAKE_SYSTEM_PROCESSOR x86_64)
+    endif()
+  endif()
+endif()
+
+set(CMAKE_SYSROOT "${_prefix}${CMAKE_SYSTEM_PROCESSOR}-w64-mingw32")
+
+if(NOT "$ENV{PKG_CONFIG_LIBDIR}" STREQUAL "")
+  set(_respath "${CMAKE_SYSROOT}/usr/lib/pkgconfig:${CMAKE_SYSROOT}/usr/share/pkgconfig:$ENV{PKG_CONFIG_LIBDIR}")
```

Comment:
There's no `/usr/lib` under the toolchain sysroots. Then again, there's no pkg-config files there at all. It would be quite unusual to have the user install the user built libraries under the toolchain sysroots too (and even then, placing it under `<sysroot>/usr` also feels a bit unexpected as we don't have any preexisting `usr` directory there).

Reply:
True, copy-pasta from my other toolchains. Gonna fix it

There are no pkg-config files, but the idea is that one could for example take this toolchain, and build QT - itself requiring dependencies that are looked up. Means you need the dependencies before and add them, Layout would be like https://packages.debian.org/sid/all/libz-mingw-w64-dev/filelist.

(and even if there never be any pkg-config files - you should still configure the variables to prevent host packages to be picked up)

Reply:
The sysroot I guess is an attempt to make it work with the --with-sysrooot=dir option. I often use this when crosscompiling applications, to have for example multiple mingw versions for x64 win, x32 win on 1 build server

Reply:
> The sysroot I guess is an attempt to make it work with the --with-sysrooot=dir option. I often use this when crosscompiling applications, to have for example multiple mingw versions for x64 win, x32 win on 1 build server

Its `--sysroot=dir` and it has 2 uses:

1.   CMake will locate libraries/includes/packages there (exclusively, ignoring host system packages)
2.   tools using `compile_commands.json` like clangd will see this option and will then use the same paths as the compiler (clangd is usually not from this toolchain and doesnt know all its defaults)

But the point was `PKG_CONFIG_LIBDIR` and this is another option to ensure 1.